src/Infrastructure.VS/Roslyn/IEmbeddedRoslynAnalyzerProvider.cs

```diff
@@ -26,7 +26,7 @@ namespace SonarLint.VisualStudio.Infrastructure.VS.Roslyn;
 public interface IEmbeddedRoslynAnalyzerProvider
 {
     /// <summary>
-    /// Returns SonarAnalyzer.CSharp & SonarAnalyzer.VisualBasic analyzer DLLs that are embedded in the VSIX
+    /// Returns SonarAnalyzer.CSharp & SonarAnalyzer.VisualBasic analyzer DLLs that are embedded in the VSIX or null if no analyzers are found.
```

Comment:
Not sure it's a good idea. I'm leaning towards throwing. Plus, as it is a contract change, you should've discussed that prior to committing to it

Reply:
They're inherently different. While for the CM provider `null` is an acceptable value (when there's no analyzer), for the embedded one it's an exceptional case that, IMO, should throw (there should be no reason to not have embedded analyzers). Moving responsibility for the error handling to a different class actually makes it more complicated